core/test/dataset_binary_arithmetic_test.cpp

```diff
@@ -16,7 +16,10 @@
 using namespace scipp;
 using namespace scipp::core;
 
-DatasetFactory3D datasetFactory;
```

Comment:
So it was not possible to simply declare
```C++
static DatasetFactory3D datasetFactory;
```
?

Reply:
No. The problem is that init order of static and global variables is undefined. If the factory is initalized before `Dim`, things go badly wrong. There is probably a better fix for this, I'll look into it once this is all working and merged.